### PR TITLE
TLMBLGIS-198 Fix unicode bug

### PR DESCRIPTION
**Pull request by dovek**
Due: 10/4
@dlhaines 
@lsloan

### CODE REVIEW

**Comment by dlhaines** [comment]
Can this be documented / explained under the Best Practices (or other appropriate and findable) wiki page?  It looks magical and arbitrary to the untrained eye.  E.g. what about this actually does anything about unicode?  Why would unicode processing need to be explicitly applied to a specific message rather than all messages? (Line 171).  Does that mean every message needs to have unicode processing applied specifically to it?

File: mbof/models.py [hunk 1]
```diff
@@ -120,8 +117,8 @@ def maybe(self):
         )
         return guestTotal
 
-    def __str__(self):
-        return str(self.eventText) + ' (' + self.__class__.__name__ + ': ' + str(self.id) + ')'
+    def __unicode__(self):
+        return '%s %s %s' % (self.eventText, self.__class__.__name__, self.id)
```
Comment (lsloan) [inline mbof/models.py:121]: Using `format()` for substitution:

``` python
'{self.eventText} {self.__class__.__name__} {self.id}'.format(**locals())
```

It's longer than the %-substitution version, but it's more readable.  Some IDEs support symbol suggestion/completion within strings when using format.

**Comment by lsloan** [comment]
The best practice that needs to be documented is that our applications must support Unicode.  Not only because people like to use emojis or other special symbols in their text (I do), we support students and faculty from a wide range of cultures that are likely to use non-ASCII characters in their text.  It would be bad to corrupt that text.

**Review by lsloan — APPROVED** [review]
I like that along with the str/unicode changes you cleaned up the code with %-substitution.  I've recently started using `format()` for this kind of substitution.  You should consider using it, too.  I'll give a specific example in my other comments.

File: mbof/models.py [hunk 2]
```diff
@@ -21,7 +21,6 @@ def currentUserObject():
     return User.objects.get(loginName=os.getenv('REMOTE_USER'))
 
 
-@python_2_unicode_compatible
```
Comment (lsloan) [inline mbof/models.py:24]: If you remove `@python_2_unicode_compatible` for this method, don't you need to rename `__str__()` to `__unicode__()` and update its code?

Comment (dovek) [reply mbof/models.py:24]: Yes, but user is being removed in a later JIRA, so I'm ignoring it for now.

**Comment by lsloan** [comment]
WRT @dlhaines questions:

Django expects output to support Unicode by default.  It complains if classes don't have a `__unicode__()` method.  The `@python_2_unicode_compatible` class decorator automatically adds that method to classes, which basically calls the `__str__()` method.

The Ocellus code was trying to get string-like representations from some instances of Event, Vote, etc. and the `__str__()` method was being called automatically, apparently bypassing the generated `__unicode__()` method.  I think Django specifically requests a Unicode string, but in these cases, other code within Ocellus was requesting the string representation without requiring Unicode support.  When that other code returned the string to Django, it reported problems.  This change ensures the strings are handled properly as Unicode.

File: mbof/models.py (hunk 2 above)
Comment (lsloan) [reply mbof/models.py:24]: But the `Role` class still needs to be updated.

Comment (dovek) [reply mbof/models.py:24]: Ah, good call. We haven't decided what we are doing with Role yet, but I will update it to unicode in the meantime. Thanks!

**Pull request closed without merging**